pkg/utils/cmd/tiller-hack/tiller-hack.go

```diff
@@ -0,0 +1,182 @@
+}
+
+func decodeRelease(data []byte) (*rspb.Release, error) {
+	// base64 decode string
+	b, err := b64.DecodeString(string(data))
+	if err != nil {
+		return nil, err
+	}
+
+	if bytes.Equal(b[0:3], magicGzip) {
+		r, err := gzip.NewReader(bytes.NewReader(b))
+		if err != nil {
+			return nil, err
+		}
+		b2, err := ioutil.ReadAll(r)
+		if err != nil {
+			return nil, err
+		}
+		b = b2
+	}
+
+	var rls rspb.Release
+	// unmarshal protobuf bytes
+	if err := proto.Unmarshal(b, &rls); err != nil {
+		return nil, err
+	}
+	return &rls, nil
+}
+
+func encodeRelease(rls *rspb.Release) (string, error) {
+	b, err := proto.Marshal(rls)
+	if err != nil {
+		return "", err
+	}
+	var buf bytes.Buffer
+	w, err := gzip.NewWriterLevel(&buf, gzip.BestCompression)
+	if err != nil {
+		return "", err
+	}
+	if _, err = w.Write(b); err != nil {
+		return "", err
+	}
+	w.Close()
+
+	return b64.EncodeToString(buf.Bytes()), nil
+}
+
+func updateConfigMap(k8sClient *kubernetes.Clientset, configMapName, configMapNamespace, updatedData string) error {
+	cfgMap, err := k8sClient.Core().ConfigMaps(configMapNamespace).Get(configMapName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("fail to get configmap %s/%s with error: %+v", configMapNamespace, configMapName, err)
+	}
+	cfgMap.Data["release"] = updatedData
+
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() (err error) {
+		_, err = k8sClient.CoreV1().ConfigMaps(configMapNamespace).Update(cfgMap)
+		return
+	})
+	return nil
+}
+
+func main() {
+	flag.Parse()
+	flag.Set("logtostderr", "true")
+
+	k8sClient, err := BuildClient(*kubeconfig)
+	if err != nil {
+		logrus.Errorf("Failed to create a client: %+v\n", err)
+		os.Exit(1)
+	}
+	if *configMapName == "" {
+		logrus.Errorf("configmap name is missing.")
+		os.Exit(1)
+	}
+	if *configMapNamespace == "" {
+		*configMapNamespace = v1.NamespaceDefault
+	}
+	data, err := getConfigmapData(k8sClient, *configMapName, *configMapNamespace)
+	if err != nil {
+		logrus.Errorf("failed to get Configmap's Data with error: %+v", err)
+		os.Exit(1)
+	}
+	release, err := decodeRelease(data)
+	if err != nil {
+		logrus.Errorf("Fail to decode Release Data of configmap with error: %+v", err)
+		os.Exit(1)
+	}
+	var updatedTemplate []*chart.Template
+	for _, template := range release.Chart.Templates {
+		if template.Name == "templates/crds.yaml" || template.Name == "templates/install-custom-resources.sh.tpl" {
+			continue
+		}
+		updatedTemplate = append(updatedTemplate, template)
+	}
+	release.Chart.Templates = updatedTemplate
+	updatedData, err := encodeRelease(release)
+	if err != nil {
+		logrus.Errorf("Fail to encode Release Data of configmap with error: %+v", err)
+		os.Exit(1)
+	}
+	if err := updateConfigMap(k8sClient, *configMapName, *configMapNamespace, updatedData); err != nil {
+		logrus.Errorf("Fail to update configmap with error: %+v", err)
+		os.Exit(1)
+	}
+
+	logrus.Infof("All done, tiller should be restarted to refresh its data store...")
```

Comment:
is it possible to automatically bounce tiller by default (and provide a flag not to do this?)  If not, can you add the command the operator should run to do the job...

Reply:
I do not want to do it as a default behaviour. I think it is too intrusive. I could add a flag:
`--tiller-restart=true` but still I am not sure if it should be done. I would like to hear from other people about this.

Reply:
such a flag seems to make good sense to me.  Having operators track down the tiller pod and bounce it may seem unnatural.